packages/dds/tree/src/feature-libraries/optional-field/optionalFieldChangeTypes.ts

```diff
@@ -61,5 +62,14 @@ export interface OptionalChangeset<TChildChange = NodeChangeset> {
 	 * However, if this changeset is then rebased over a change which populates `foo`, the rebased changeset must now empty `foo`.
 	 * This reserved id is used as the destination of that emptying move.
 	 */
-	reservedDetachId?: RegisterId;
+	readonly field?: Replace;
+}
+
+export interface Replace {
```

Comment:
could you document this? the property `isEmpty` also makes me think maybe the conditions around `field` above might need to be updated (maybe condition 2 is no longer required? not sure without understanding semantics of this, though)

Reply:
based on other contents of the PR, it looks like representation changes here are:

- Moves in/out of the "self" register (which previously were part of `moves` above) are now no longer part of `moves` and instead are represented here.
- `src`, when populated, indicates the register which this change moves into the contents of the field
- `dst` always represents the change atom id which field contents should be moved to, should this change be applied in an input context where the field has content
- `isEmpty` dictates whether the input context has content